packages/template-retail-react-app/app/components/unavailable-product-confirmation-modal/index.jsx

```diff
@@ -38,21 +39,25 @@ const UnavailableProductConfirmationModal = ({
                 const resProductIds = []
                 const unOrderableIds = []
                 result.data?.forEach(({id, inventory}) => {
-                    const productItem = productItems.find((item) => item.productId === id)
-                    // wishlist item will have the property type
-                    const isWishlist = !!productItem?.type
                     // when a product is unavailable, the getProducts will not return its product detail.
                     // we compare the response ids with the ones in basket to figure which product has become unavailable
                     resProductIds.push(id)
+
                     // when a product is orderable, but the quantity in the basket is more than the remaining stock
                     // we want to make sure it is removed before go to checkout page to avoid error when placing order
                     // we don't need to remove low stock/ out of stock from wishlist
-                    if (
-                        !isWishlist &&
-                        (!inventory?.orderable ||
-                            (inventory?.orderable && productItem?.quantity > inventory.stockLevel))
-                    ) {
-                        unOrderableIds.push(id)
+                    if (productItems.length) {
```

Comment:
Edge case: if both productIds and productItems are passed in, we prioritize using productIds as it is the older behavior that will not break existing implementaions.

But here the condition is to run the extra logic if productItems have been sent in. Should this also have a condition where we don't run this block if productIds have length?

So rather than `if (productItems.length)` we use `if (productItems.length && !productIds.length)`?

Reply:
if productItems is passed, it takes precedency, I don't think we need to care about productIds at this point. but you have a good point I find adding a warning here to let users know they are not supposed to pass 2 props at the same time 🤔

Reply:
Good point, I think here we should have a warning to not allow two props being passed at the same time. Thoughts?

Reply:
On second thought, if they pass in both productIds and productItem, they will have extra behavior where low stock /out of stock products in basket being removed depends on the productIds that they pass in. I think it should be fine. 🤔